### ISSUE TITLE
Cannot get correct union case field names using reflection

### ISSUE THREAD

**Issue by anchann** [fable-compiler/Fable#2017]
### Description

When using reflection APIs, the name reported for a union case field seems to be some kind of autogenerated name instead of the real one in the type definition. How can I get the real name?

### Repro code

```
open System
open FSharp.Reflection

type Person = Person of FirstName: string * LastName: string

let fieldNames: seq<string> =
    FSharpType.GetUnionCases typeof<Person>
    |> Seq.collect (fun caseInfo ->
        caseInfo.GetFields()
        |> Seq.map (fun field ->
            sprintf "%s or %s" field.PropertyType.Name field.Name
        )
    )

printf "%O" fieldNames
```

### Expected and actual results

Running this in the fable REPL produces

```
seq [String or Data0; String or Data1]
```

Running it in `fsi` produces the expected

```
seq ["String or FirstName"; "String or LastName"]
```

**Comment by inosik** [comment]
I'll leave some links here for anybody who's willing to tackle this:

- https://github.com/fable-compiler/Fable/blob/fa8d44f6cff1a1446078b84f9f1c72979d5fd9d5/src/fable-library/Reflection.ts#L13
  `CaseInfo.fields` should be updated to be an array of `FieldInfo`

- https://github.com/fable-compiler/Fable/blob/fa8d44f6cff1a1446078b84f9f1c72979d5fd9d5/src/Fable.Transforms/Fable2Babel.fs#L374
  The F# side of the prior function should be updated as well to build an array of `FieldInfo` objects. Must be this snippet:
  https://github.com/fable-compiler/Fable/blob/fa8d44f6cff1a1446078b84f9f1c72979d5fd9d5/src/Fable.Transforms/Fable2Babel.fs#L390-L393

- https://github.com/fable-compiler/Fable/blob/fa8d44f6cff1a1446078b84f9f1c72979d5fd9d5/src/fable-library/Types.ts#L129
  `Union` type in `Types.ts` might need to be updated too

**Comment by anchann** [comment]
@inosik thank you for the guidance. I'd like to work on this as I need the fix asap for some production-facing work I'm doing, but couldn't get a local build of Fable going. Any chance you could help me with https://github.com/fable-compiler/Fable/issues/2018?

**Comment by anchann** [comment]
Figured out #2018. Will proceed trying to implement this.

**Comment by MangelMaxime** [comment]
It has been fixed in fable-compiler 2.6.0
